### PR TITLE
lightning: add more retryable err (#36673)

### LINKED ISSUE #36674: lighting import failed while import on DBAAS
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. create 1* TiDB, 3*Tikv 2u8G cluster.
2. try to import 100GB CSV file

### 2. What did you expect to see? (Required)
import success

### 3. What did you see instead (Required)
import fail,  and report Root cause: [Lightning:Restore:ErrRestoreTable]restore table `gotpc_1500`.`order_line` failed: EOF; EOF; EOF.

### 4. What is your TiDB version? (Required)

v6.1.0

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #36673 to release-6.1
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/36719
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/36719:release-6.1-bf706ac12dd7
```

---

### What problem does this PR solve?

Issue Number: close #36674

Problem Summary:

### What is changed and how it works?
add retry on `io.EOF` and `coprocessor task terminated due to exceeding the deadline` and `Target table is calculating checksum. Please wait until the checksum is finished and try again.`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/meta_manager.go [hunk 1]
```diff
@@ -179,124 +185,149 @@ func (m *dbTableMetaMgr) AllocTableRowIDs(ctx context.Context, rawRowIDMax int64
...
-			if status >= metaStatusFinished {
-				continue
-			}
+				// skip finished meta
+				if status >= metaStatusFinished {
+					continue
+				}
 
-			if status == metaStatusChecksuming {
-				return common.ErrAllocTableRowIDs.GenWithStack("Target table is calculating checksum. Please wait until the checksum is finished and try again.")
```
Comment (D3Hunter) [inline br/pkg/lightning/restore/meta_manager.go:212]: only this line is changed, other lines only changes indents

Comment (D3Hunter) [reply br/pkg/lightning/restore/meta_manager.go:212]: this pr is edited to use this part of code in 6.1, so it's line count changed is different with origin pr

**Review by buchuitoudegou — APPROVED** [review]

**Review by dsdashun — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
ci pass, merge
https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/12764/pipeline

**Pull request merged**